### PR TITLE
Fix #82 -- Handle bad request errors and feed them back to the user

### LINKED ISSUE #82: BadRequestError: Error code: 400 - {'error': {'message': 'Files with extensions [.png] are not supported for retri...
Sentry Issue: [VOIIO-SAM-P](https://voiio.sentry.io/issues/5236434175/?referrer=github_integration)

```
BadRequestError: Error code: 400 - {'error': {'message': 'Files with extensions [.png] are not supported for retrieval. See https://platform.openai.com/docs/assistants/tools/supported-files', 'type': 'invalid_request_error', 'param': None, 'code': 'unsupported_file'}}
(3 additional frame(s) were not displayed)
...
  File "sam/slack.py", line 82, in handle_message
    has_attachments, has_audio = await bot.add_message(
  File "sam/bot.py", line 248, in add_message
    await client.beta.threads.messages.create(

Failed to run listener function (error: Error code: 400 - {'error': {'message': 'Files with extensions [.png] are not supported for retrieval. See https://platform.openai.com/docs/assistants/tools/supported-files', 'type': 'invalid_request_error', 'param': None, 'code': 'unsupported_file'}})
```

### PR DESCRIPTION
**Pull request by herrbenesch**
image

### How to test
- Setup alpha sam
1. Upload a png file in a private chat
2. Upload a pdf file in a private chat and add no message

Open question
- Does currently not work within a group chat. Even if sam is addressed directly. I'm not sure we need this. And if we need it, I'm not sure how to address it.

### CODE REVIEW

File: sam/bot.py [hunk 1]
```diff
@@ -236,15 +236,22 @@ async def add_message(
...
+            role=Roles.USER,
+            attachments=[
+                {"file_id": file_id, "tools": [{"type": "file_search"}]}
+                for file_id in file_ids
+            ],
+        )
+    except openai.BadRequestError as e:
+        error = OSError()
+        error.strerror = e.body["message"]
+        raise error
```
Comment (codingjoe) [inline sam/bot.py:253]: You are swallowing the original stack trace.
```suggestion
    except openai.BadRequestError as e:
        raise OSError("A proper message") from e
```

It's also a bit strange to capture a more specific error and to replace it with a less descriptive one.

File: sam/slack.py [hunk 2]
```diff
@@ -77,11 +77,20 @@ async def handle_message(event: {str, Any}, say: AsyncSay):
...
+                files=files,
+            )
+        except OSError as e:
+            # Only relevant for direct messages
+            if channel_type == "im":
+                await say(
+                    channel=channel_id,
+                    text=f"I'm sorry, I can't process this. {e.strerror}",
+                )
+            return
```
Comment (codingjoe) [inline sam/slack.py:93]: We usually just post this 🤯. You might also want to add a warning log with `exc_info=True` to send the stack trace to sentry.

File: sam/bot.py (hunk 1 above)
Comment (herrbenesch) [reply sam/bot.py:253]: I'll give this a shot. My overall goal was to pass the most useful bit of information back to the user.

**Comment by herrbenesch** [comment]
@codingjoe thanks for the review. I'm now feeding back the error messages to the assistant in order to receive a nicely formatted issue description.

**Review by codingjoe — APPROVED** [review]
Thanks for addressing the comments. Interesting approach to use the not to translate the error into a human readable Output.
Let's give it a try

**Comment by herrbenesch** [comment]
thanks @codingjoe

**Pull request merged**